js/popup.js

```diff
@@ -0,0 +1,61 @@
+import { TYPES_OF_HOUSING, generateData } from './generate-data.js';
+
+const offerTemplate = document.querySelector('#card').content.querySelector('.popup');
+
+//creating new ads BY HTML TEMPLATE
+const getPopup = (ad) => {
+  const { title, address, price, type, rooms, guests, checkin, checkout, features, description, photos } = ad.offer;
+  const { avatar } = ad.author;
+
+  const adItem = offerTemplate.cloneNode(true);
+
+  adItem.querySelector('.popup__title').textContent = (title)
+    ? title
+    : adItem.querySelector('.popup__title').remove();
```

Comment:
```suggestion
  adItem.querySelector('.popup__title').textContent = (title)
    ? title
    : adItem.querySelector('.popup__title').remove();
```
два раза идет поиск одного и того же элемента. лучше это вынести в одну переменную
как вариант, саму проверку можно вынеси в функции, а то она всегда повторяется